src/state/pbridge/hooks.ts

```diff
@@ -372,10 +391,135 @@ export function useBridgeSwapActionHandlers(): {
+        toChain: toChain?.chain_id || 1,
+        toToken:
+          toCurrency?.address
+            ?.toString()
+            ?.replace('0x0000000000000000000000000000000000000000', '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') || '',
+        toAddress: recipient || fromAddress || '0x0000000000000000000000000000000000000000',
+        slippage: parseFloat(slipLimit),
+        enableForecall: true,
+      };
+
+      let squidRouteRes: RouteResponse | null;
+      try {
+        const squidRoutes = await axios.get(`${SQUID_API}/v1/route`, {
+          params: params,
+        });
+        squidRouteRes = squidRoutes.data;
+      } catch (error) {
+        squidRouteRes = null;
+      }
+
+      const squidRoute: Route | undefined = squidRouteRes?.route
+        ? {
+            nativeRoute: squidRouteRes.route as SquidRouteData,
+            bridgeType: SQUID,
+            waitingTime: calculateTransactionTime(squidRouteRes.route.estimate.estimatedRouteDuration),
+            toToken: toCurrency?.symbol || '',
+            toAmount: new TokenAmount(toCurrency as Currency as Token, squidRouteRes.route.estimate.toAmount).toFixed(
+              4,
+            ),
+            toAmountNet: new TokenAmount(
+              toCurrency as Currency as Token,
+              squidRouteRes.route.estimate.toAmountMin,
+            ).toFixed(4),
+            toAmountUSD: `${squidRouteRes.route.estimate?.toAmountUSD} USD`,
+            gasCostUSD: squidRouteRes.route.estimate.gasCosts
+              .reduce((prevValue, currentValue) => {
+                return prevValue + parseFloat(currentValue.amountUSD);
+              }, 0)
+              .toFixed(2),
+            steps: [
+              {
+                bridge: SQUID,
+                type: 'bridge',
+                includedSteps: [
+                  ...squidStepGenerator(
+                    squidRouteRes.route.estimate.route.fromChain,
+                    toCurrency,
+                    squidRouteRes.route.estimate.toAmount,
+                  ),
+                  ...squidStepGenerator(
+                    squidRouteRes.route.estimate.route.toChain,
+                    toCurrency,
+                    squidRouteRes.route.estimate.toAmount,
+                  ),
+                ],
+                action: {
+                  toToken: toCurrency?.symbol || '',
+                },
+                estimate: {
+                  toAmount: new TokenAmount(
+                    toCurrency as Currency as Token,
+                    squidRouteRes.route.estimate.toAmount,
+                  ).toFixed(4),
+                },
+              },
+            ],
+            transactionType: BridgePrioritizations.RECOMMENDED,
+            selected: false,
+          }
+        : undefined;
+
+      dispatch(
+        setRoutes({
+          routes: [...lifiRoutes, squidRoute].filter((x: Route | undefined) => x !== undefined) as Route[],
+          routesLoaderStatus: false,
+        }),
+      );
```

Comment:
1. `getRoutes` function started getting larger now. lets split bridge wise functions. like 
`getLifiRoutes`, `getSquidRoutes` then in `getRoutes` we just need to call those functions and merge those results in `routes`
2. here in `filter` you can do `.filter((x: Route | undefined) => !!x)`

Reply:
1. Yeah, I was planning to do this in a different PR. Because need to add different Loader states etc.
2. Done